opencl-aot/source/main.cpp

```diff
@@ -426,9 +556,30 @@ int main(int Argc, char *Argv[]) {
   }
   std::cout << "Using build options: " << BuildOptions << '\n';
 
-  // step 8: build OpenCL program
-  CLErr = clBuildProgram(ProgramUPtr.get(), 1, &DeviceId, BuildOptions.c_str(),
-                         nullptr, nullptr);
+  // step 8: compile | build | link OpenCL program
+  if (OptCommand.getNumOccurrences() && OptCommand == Commands::compile) {
+    assert(OptDevice == fpga_fast_emu &&
+           "Use -cmd=compile with --device=fpga_fast_emu only");
+    CLErr =
+        clCompileProgram(ProgramUPtr.get(), 1, &DeviceId, BuildOptions.c_str(),
+                         0, nullptr, nullptr, nullptr, nullptr);
+  } else if (OptCommand.getNumOccurrences() && OptCommand == Commands::link &&
+             OptBinaries.getNumOccurrences()) {
+    assert(OptDevice == fpga_fast_emu &&
+           "Use -cmd=link with --device=fpga_fast_emu only");
```

Comment:
Same question as above about this assertion

Reply:
Removed.